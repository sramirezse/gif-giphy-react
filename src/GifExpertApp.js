import React from 'react'
import { useState } from 'react'
import { CategoryAdd } from './components/CategoryAdd'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Dragon ball'])
    // const handleAdd =  () => {
    //     setCategories( cats=> ['otra serie',...cats])
    // }
    return (
        <div>
            <h2 className='animate__animated animate__lightSpeedInRight'> GifExpertAppJS </h2>
            <CategoryAdd setCategories={setCategories} />
            <hr></hr>

            {
                categories.map(category => (
                    <GifGrid category={category} key={category} />
                ))
            }

        </div>
    )
}
