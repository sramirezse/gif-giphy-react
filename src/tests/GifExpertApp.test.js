import React from 'react';
import { GifExpertApp } from "../GifExpertApp";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";


describe('Pruebas en <GifExpertApp />',()=>{
    test('Snapshot', ()=>{
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();


    });
});