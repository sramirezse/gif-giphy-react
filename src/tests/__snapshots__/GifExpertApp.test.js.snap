// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Snapshot 1`] = `
<div>
  <h2
    className="animate__animated animate__lightSpeedInRight"
  >
     GifExpertAppJS 
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Dragon ball"
    key="Dragon ball"
  />
</div>
`;
